pg_search/src/postgres/storage/linked_items.rs

```diff
@@ -140,6 +142,17 @@ impl<T: From<PgItem> + Into<PgItem> + Debug + Clone + MVCCEntry> LinkedItemList<
                 if let Some(entry) = page.read_item::<T>(offsetno) {
                     if entry.recyclable(snapshot, heap_relation) {
                         delete_offsets.push(offsetno);
+                    } else {
+                        let xmin_needs_freeze = entry.xmin_needs_freeze(freeze_limit);
+                        let xmax_needs_freeze = entry.xmax_needs_freeze(freeze_limit);
+
+                        if xmin_needs_freeze || xmax_needs_freeze {
+                            let frozen_entry =
+                                entry.into_frozen(xmin_needs_freeze, xmax_needs_freeze);
+                            let PgItem(item, size) = frozen_entry.clone().into();
+                            let did_replace = page.replace_item(offsetno, item, size);
+                            assert!(did_replace);
```

Comment:
I won't let this hold things up, but I wonder if `page.replace_item()` could ever fail for this item?  I suppose if one of T's members is an integer and bincode variable-width encodes them, then it _could_ change size with the new values?  I dunno.

Again, don't think we should hold things up -- if it ever happens we'll know why.

Reply:
I don't think so. Because the only thing we're changing about the struct is the `u32` xmax, the size should stay the same.